src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -5635,6 +5661,113 @@ public void Utf8Output()
             parsedArgs.Errors.Verify(Diagnostic(ErrorCode.ERR_BadSwitch).WithArguments("/utf8output:"));
         }
 
+        [Fact]
+        public void CscCompile_WithSourceCodeRedirectedViaStandardInput_ProducesRunnableProgram()
+        {
+            ProcessResult result = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                ProcessUtilities.Run("cmd", $@"/C echo  ^
+class A                                                 ^
+{{                                                      ^
+    public static void Main() =^^^>                     ^
+        System.Console.WriteLine(""Hello World!"");     ^
+}} | {s_CSharpCompilerExecutable} /nologo /t:exe -"
+    .Replace(Environment.NewLine, string.Empty)) :
+                ProcessUtilities.Run("/usr/bin/env", $@"sh -c ""echo  \
+class A                                                               \
+{{                                                                    \
+    public static void Main\(\) =\>                                   \
+        System.Console.WriteLine\(\\\""Hello World\!\\\""\)\;         \
+}} | {s_CSharpCompilerExecutable} /nologo /t:exe -""",
+                    // we are testing shell's piped/redirected stdin behavior explicitly
+                    // instead of using Process.StandardInput.Write(), so we set
+                    // redirectStandardInput to true, which implies that isatty of child
+                    // process is false and thereby Console.IsInputRedirected will return
+                    // true in csc code.
+                    redirectStandardInput: true);
+
+            Assert.False(result.ContainsErrors, $"Compilation error(s) occurred: {result.Output} {result.Errors}");
+
+            string output = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                ProcessUtilities.RunAndGetOutput("cmd.exe", $@"/C ""{s_DotnetCscRun} stdin.exe""", expectedRetCode: 0) :
```

Comment:
>stdin.exe [](start = 85, length = 9)

Where is this file created?

Reply:
Current working directory of test: `artifacts\bin\Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests\{Configuration}\{TFM}`.

Reply:
Please consider creating a temp directory for each test. Something like:
```C#
var path = Temp.CreateDirectory().Path;
...
ProcessUtilities.RunAndGetOutput("cmd.exe", ..., startFolder: path);

```

---
In reply to: [373214597](https://github.com/dotnet/roslyn/pull/41166#discussion_r373214597) [](ancestors = 373214597)